CaseBonita/Services/PlaylistUploader/Handlers/Spotify.py

```diff
@@ -1,26 +1,77 @@
+import json
+import webbrowser
+
+import requests
 import spotipy as sp
 from spotipy.oauth2 import SpotifyClientCredentials
 
 from CaseBonita.Services.PlaylistUploader.Handlers.Base import BaseAPIUploader
-from .consts import CLIENT_ID, CLIENT_SECRET
+from .consts import CLIENT_ID, CLIENT_SECRET, SPOTIFY_USER_ACCESS
+from .utils import insert_id_to_url
 
 
 class UploadToSpotify(BaseAPIUploader):
     def __init__(self, developer_key, _songs_json_file, user_credentials):
         super(BaseAPIUploader, self).__init__(developer_key, _songs_json_file, user_credentials)
 
-    def create_developer_credentials(self):
+    @classmethod
+    def return_developer_credentials(cls):
         credentials_manager = SpotifyClientCredentials(client_id=CLIENT_ID, client_secret=CLIENT_SECRET)
-        self.developer_key = sp.Spotify(client_credentials_manager=credentials_manager)
-        return self.developer_key
-
-    def get_tracks_id(self, artist, track):
-        track_id = self.developer_key.search(q='artist:' + artist + ' track:' + track, type='track')
-        self.songs_json_file = self.songs_json_file.update('track_id':track_id)
-        return self.songs_json_file
-
-    def create_playlist(self, playlist_name):
-        token = self.developer_key.user_playlist_create(user=self.user_credentials['username'],
-                                                        name=playlist_name,
-                                                        public=True)
-        return token
+        cls.developer_key = sp.Spotify(client_credentials_manager=credentials_manager)
+        return cls.developer_key
+
+    @classmethod
+    def get_tracks_id(cls, artist, track):
+        track_id = cls.developer_key.search(q='artist:' + artist + ' track:' + track, type='track')
+        cls.songs_json_file = cls.songs_json_file.update('track_id':track_id)
+        return cls.songs_json_file
+
+    @classmethod
+    def generate_access_token(cls):
+        """
+        Have the user login with spotify and return the auth token generated by spotify for him.
+        :return: str user token:
+
+        Currently set access is via local server set by the beast, spotify requires a redirect URI in order to
+        generate token.
+
+        Base access is hour long.
+
+        """
+        new = 2
```

Comment:
why 2?

Reply:
opens the browser on a new tab.

Reply:
if you decide to keep this piece of code, create a const for something like that - even an enum of codes that represent functionality. 
Furthermore, selenium itself may have const for this, worth to check.